rails/app/controllers/search_controller.rb

```diff
@@ -313,7 +313,9 @@ def get_current_material_unassigned_collections
     @unassigned_collections = @collections - @assigned_collections
 
     @skip_reload = params[:skip_reload] == 'true'
-    render :partial => 'material_unassigned_collections'
+    respond_to do |format|
```

Comment:
Since we are cleaning this up. The name `get_current_material_unassigned_collections` doesn't make as much sense anymore.

I'd suggest moving this action and the `add_material_to_collections` action into the ExternalActivities controller. The names could be `edit_collections` and `update_collections`.   
The route for each could be `collections`: edit_collections is a GET and update_collections is a POST. So I think the routes would be:
- `match 'collections' => 'edit_collections', via: :get`
- `match 'collections' => 'update_collections', via: :post`

The authorization to do this editing and updating will need to be updated. Right now these two actions are limited by this line at the top of the file: 
`before_filter :admin_only, :only => [:get_current_material_unassigned_collections, :add_material_to_collections]
`

In the ExternalActivitiesController pundit is used so `authorize @external_activity` should be called and new methods need to be added to the external_activity_policy.rb should restrict this to just admins.